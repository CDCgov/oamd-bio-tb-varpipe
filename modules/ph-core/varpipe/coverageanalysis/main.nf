process VARPIPE_COVERAGEANALYSIS {

    tag "${meta.id}"
    label 'process_single'

    container 'oamd-bio-biocontainers-python:3.12_80c558c_v3'

    input:
    tuple val(meta), path(coverage), path(unmapped), path(mapped)
    path(amp) // gene regions (single BED file)
    path(features) // feature regions (multiple BED files)

    output:
    tuple val(meta), path("*_stats.txt"), emit: stats
    tuple val(meta), path("*_target_region_coverage.txt"), emit: target_region_coverage, optional: true
    tuple val(meta), path("*_genome_region_coverage.txt"), emit: genome_region_coverage, optional: true
    tuple val(meta), path("*_qc_log.txt"), emit: qc_failed, optional: true
    tuple val(meta), path(".qc_passed.txt"), emit: qc_passed, optional: true
    path("versions.yml"), emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    template 'coverage_stats.py'

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_stats.txt
    touch ${prefix}_target_region_coverage.txt
    touch ${prefix}_genome_region_coverage.txt
    touch ${prefix}_structural_variants.txt
    touch .qc_passed.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | cut -d' ' -f2)
    END_VERSIONS
    """
}
