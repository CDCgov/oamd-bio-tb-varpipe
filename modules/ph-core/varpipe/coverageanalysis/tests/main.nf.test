nextflow_process {
    name "Test Process VARPIPE_COVERAGEANALYSIS"
    script "../main.nf"
    process "VARPIPE_COVERAGEANALYSIS"

    tag "modules"
    tag "modules_phcore"
    tag "varpipe"
    tag "varpipe/coverageanalysis"

    test("varpipe coverage analysis [bam] [coverages]") {
        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test_coverage.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample1_unmapped.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample1_mapped.txt')
                ]

                input[1] = file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/amp_bed.txt')

                input[2] = [
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/bed_1.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/bed_2.txt')
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.target_region_coverage,
                    process.out.genome_region_coverage,
                    process.out.versions
                ).match() }
            )
        }
    }

    test("varpipe coverage analysis [bam] [coverages] -stub") {
        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test_coverage.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample1_unmapped.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample1_mapped.txt')
                ]

                input[1] = file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/amp_bed.txt')

                input[2] = [
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/bed_1.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/bed_2.txt')
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.target_region_coverage,
                    process.out.genome_region_coverage,
                    process.out.versions
                ).match() }
            )
        }
    }
}
