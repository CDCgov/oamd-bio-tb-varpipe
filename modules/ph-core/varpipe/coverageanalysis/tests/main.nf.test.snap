{
    "varpipe coverage analysis [bam] [coverages] -stub": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test_target_region_coverage.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test_genome_region_coverage.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "versions.yml:md5,817cdee52b21f086e52eac7a69051381"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.1"
        },
        "timestamp": "2025-05-19T15:33:35.835411719"
    },
    "varpipe coverage analysis [bam] [coverages]": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test_target_region_coverage.txt:md5,d7d04cff5b7194ed4ac7888773c25eca"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test_genome_region_coverage.txt:md5,8ee6f9a8b985ed77c168e553df461ea1"
                ]
            ],
            [
                "versions.yml:md5,f17eed4892f91622a93fcc6e8efc32fc"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.1"
        },
        "timestamp": "2025-05-19T15:33:29.149412347"
    }
}
